### DISCUSSION TITLE
[Task]: Robust testing of temporal averaging APIs (including missing data) with real world datasets

### DISCUSSION THREAD

**Discussion by tomvothecoder** [xCDAT/xcdat#354] [4. General]
This post keeps a dedicated track record for testing temporal averaging APIs with real world datasets. 
The discussion stems from [this comment](https://github.com/xCDAT/xcdat/discussions/275#discussioncomment-3511567) in #275, which includes some temporal averaging testing by @lee1043.

We have some notebooks here for earlier versions of `xcdat` that test using a few datasets from [PMP demo data](https://github.com/PCMDI/pcmdi_metrics/blob/main/doc/jupyter/Demo/Demo_0_download_data.ipynb):

* https://github.com/xCDAT/xcdat_test/tree/main/validation/v0.2.0/temporal_average
* https://github.com/xCDAT/xcdat_test/tree/main/validation/v1.0.0/temporal_average -- note, not actually v1.0.

However, more robust testing is needed by looping over models and ensuring edge cases are covered (e.g., missing data).

FYI @lee1043, @msahn, @pochedls.

**Comment by tomvothecoder** [comment]
0.3.3 was just released which includes #350, drop leap days for specific CF calendar types for daily climatology and departures.

@lee1043, @msahn, @gleckler1, you can pull the latest version and test xcdat's temporal averaging with PMP now! Thanks!

Changelog: https://github.com/xCDAT/xcdat/discussions/359

**Reply by lee1043** [reply]
@tomvothecoder Excellent, thank you for the release!
